cpp-package/include/mxnet-cpp/ndarray.hpp

```diff
@@ -331,6 +332,16 @@ inline const mx_float *NDArray::GetData() const {
   return static_cast<mx_float*>(ret);
 }
 
+inline const mx_float *NDArray::GetDataGpu() const {
+  void *ret;
+  CHECK_EQ(GetContext().GetDeviceType(), DeviceType::kGPU);
```

Comment:
I don't think you need a new interface. You can simply remove the check above.

@lx75249